### ISSUE TITLE
Comparison to turboMaCk/any-dict

### ISSUE THREAD

**Issue by Orasund** [Timo-Weike/generic-collections#3]
Hi there,
I noticed that you did not mention [turboMaCk/any-dict](https://package.elm-lang.org/packages/turboMaCk/any-dict/latest/Dict-Any) and [turboMaCk/any-set](https://package.elm-lang.org/packages/turboMaCk/any-set/latest/) in your Readme. They are the two packages I usually use for dicts and sets. I was wondering what makes your version better to the one I'm already using?

**Comment by Timo-Weike** [comment]
Hi, as far as I can tell, besides some implementation details, the type `AnyDict` and `AnySet` from the packages you mentioned seem to be functionally equivalent to the type `Auto...` types this packages provides.
One additional different is that there are some special functions this packages does not provide but the packages for `turboMaCk` does and vice versa. 

Also my motivation for this package was and is that this package provides a generic-types solution for sets and dictionaries (and that in two fashions).

But if I'm honst I would prefer it if elm would just simply introduce the language feature of type-classes, either in general or for some predefined set that each custome type could implement (like equality, comparable, conversion to string, ...

And if this is not a plan for the language at the moment, then I would wish to have a "offical" package that is though as a work around.

tl;dr
At the moment if you are happy with the packages, keep using them. My motivation for this package was to provide one source of a workaround for a common problem.

**Comment by Timo-Weike** [comment]
As a possible future reason to switch:

I plan on implementing some further functions like the ones provided by `elm-community/dict-extra` for the core type. Furthermore do I want to implement a support for de-/encoding from/to Json.

**Comment by Orasund** [comment]
Thanks for that detailed answer. I'll definitely keep an eye out for any further additions to your package. ;)
